Applications/FileIO/SHPInterface.cpp

```diff
@@ -213,87 +213,110 @@ void SHPInterface::readPolygons(const SHPHandle& hSHP, int numberOfElements,
 
 bool SHPInterface::write2dMeshToSHP(const std::string &file_name, const MeshLib::Mesh &mesh)
 {
-    if (mesh.getDimension()!=2)
+    if (mesh.getDimension() != 2)
     {
-        ERR ("SHPInterface::write2dMeshToSHP(): Mesh to Shape conversion is only working for 2D Meshes.");
+        ERR("SHPInterface::write2dMeshToSHP(): Mesh to Shape conversion is "
+            "only working for 2D Meshes.");
         return false;
     }
 
-    unsigned nElements (mesh.getNumberOfElements());
-    if (nElements<1)
+    std::size_t const n_elements = mesh.getNumberOfElements();
+    if (n_elements < 1)
     {
-        ERR ("SHPInterface::write2dMeshToSHP(): Mesh contains no elements.");
+        ERR("SHPInterface::write2dMeshToSHP(): Mesh contains no elements.");
         return false;
     }
 
-    if (nElements>10E+7) // DBF-export requires a limit, 10 mio seems good for now
+    // DBF-export requires a limit of records, 10 mio elements seems good for now
+    std::size_t const max_exp = 8;
```

Comment:
minor: better not to mention code constants in the comment, because the comment diverges very fast from the actual code. I'd write combine the above comment with the latter one:
```c++
// DBF-export requires a limit of records because of the limits to the integer values.
// The exponent controls maximum number of elements and the maximum number of digits written in the DBF file.
```
Adjust accordingly.